users: Indicate the current user

marusak opened the pull request:
We want to obsolete 'Account settings' from the top right menu and we need to show which user is the current one.

![shortname](https://user-images.githubusercontent.com/12330670/81737122-e6b7d000-9497-11ea-8eb4-39146b055814.png)

andreasn commented:
I can't really think of anything better right now either.
Might look slightly better if you put it next to the name and allow the tiles a little more air to breathe.
I know @garrett wanted to make some bigger changes to this page, but as a first step, I think this is good enough as an incremental step.

marusak commented:
`RuntimeError: TypeError: Cannot read property 'sort' of undefined`
hmm, what have I done?

> Might look slightly better if you put it next to the name and allow the tiles a little more air to breathe.

Yeah, thought about that, but if the name is long, then the card could span a lot. Also `current` in other languages can be longer. And @mvollmer was thinking if `current` is good enough label or show we go rather with something like `your account`.

> I know @garrett wanted to make some bigger changes to this page

Oh yeah, the page could do with redesign for sure :)

andreasn commented:
I think _your account_ is a lot clearer indeed.
Agree with the reservations about the string lengths. Looks good to go for me.

garrett commented:
> Yeah, thought about that, but if the name is long, then the card could span a lot

It could wrap in that case. Either as inline (inline-block, etc.) or flex. (I'd suggest trying it out with inline / inline-block first.)


icon | User Name |  
-- | -- | --
**icon** | login | current user



Otherwise, it should _at least_ not overlap with the picture's width. It should be like this:

icon | User Name
-- | --
**icon** | login
  | current user

It could be achieved like:

[ icon ] [ info ]

...and allow info to wrap or be taller or whatever.

---

**Future 1**: Or perhaps, we could overlay it over the icon, as the icon isn't being used otherwise right now, right? (But this _might_ be considered an enhancement for later.)

**Future 2**: If we do move it to its own row, with admin accounts on their own row too, and then the rest of the accounts, it probably wouldn't even matter (unless we want to do some grid alignment).

---

Anyway, this is almost good enough for a first pass. :wink:

marusak commented:
> It could wrap in that case.

Yeah, I was wondering about this, if it would not span the box and thus make boxes of different sizes.

> Otherwise, it should at least not overlap with the picture's width. It should be like this:

Heh, that is what I started with and then decided I'll give it the full width :)

> Or perhaps, we could overlay it over the icon, as the icon isn't being used otherwise right now, right?

Right, the icon has no real purpose except makes things look better. We could surely  overlay the icon, the problem is with widths and if the string becomes too long.


I'll try it with wrapping and will see how it behaves when things become too long.

KKoukiou commented on pkg/users/local.js:
```diff
@@ -318,6 +324,8 @@ function log_unexpected_error(error) {
 PageAccounts.prototype = {
     _init: function() {
         this.id = "accounts";
+        this.permission = cockpit.permission({ admin: true });
```
I think we need to use the pkg/lib/superuser.jsx since last Marius changes.

marusak replied:
I don't think that superuser actually has user. Does it? If I saw correctly, there is only `allowed`.

marusak commented:
Ok, so here it is how it looks now for long and short name. Also changed to `Your account` and fixed bug I introduced. Tests should be happy now.

![shortname](https://user-images.githubusercontent.com/12330670/81737165-f9320980-9497-11ea-81b9-f762bb8ca61b.png)
![longlong](https://user-images.githubusercontent.com/12330670/81737166-f9caa000-9497-11ea-85f1-2cfec40f9fc4.png)
